pkg/status/health/health.go

```diff
@@ -97,6 +99,10 @@ func (c *catalog) pingComponents(healthDeadline time.Time) bool {
 	c.Lock()
 	defer c.Unlock()
 	for _, component := range c.components {
+		// In startup mode, we skip already healthy components.
+		if c.startup && component.healthy {
+			continue
```

Comment:
I think rather than modifying the catalog struct to have the `startup` field, we could added to the components struct instead.

```go
if component.startup && component.healthy
```

What do you think?

Reply:
Also, should we add a test that validate that for startup components once they are healthy we skip them?

Reply:
> I think rather than modifying the catalog struct to have the `startup` field, we could added to the components struct instead.
> 
> ```go
> if component.startup && component.healthy
> ```
> 
> What do you think?

Hey Gustavo 👋 Why would we want to have the `startup` field in the `component` struct instead of the `catalog` one?

Reply:
> Also, should we add a test that validate that for startup components once they are healthy we skip them?

Sure, I'll add this in the `TestStartupCatalog` test 👍

Reply:
Test added here: https://github.com/DataDog/datadog-agent/pull/25973/commits/d05f80667c507a1c6b092f35f627686d85fce435

Reply:
> Hey Gustavo 👋 Why would we want to have the startup field in the component struct instead of the catalog one?

The fact that we now have a flag to signal that main loop for the catalog has to change feels weird to me. The fact that the catalog only loops over components and calls the same functionality of each component. 

It seems that not all components are startup component, so the information about the type of component indicates that the `startup` field belongs to the component struct.

It might require a bit of refactoring.

One way to work with it would be to change the `register` function from catalog to rather than accepting a `name string` we should accept a component

```go
// register a component with the default 30 seconds timeout, returns a token
func (c *catalog) register(comp *component) *Handle {
	c.Lock()
	defer c.Unlock()

	if len(c.components) == 0 {
		go c.run()
	}

	h := &Handle{
		C: comp.healthChan,
	}

	// Start with a full channel to component is unhealthy until its first read
	for i := 0; i < bufferSize; i++ {
		comp.healthChan <- time.Now().Add(pingFrequency)
	}

	c.components[h] = comp
	return h
}
```

Reply:
We discussed offline, and since the same component can be register in multiple catalogs we agreed that it makes sense to leave as it is